### PR TITLE
tests: collect logs when timeout waiting for service

### PR DESCRIPTION
**Pull request by cpanato**

### CODE REVIEW

**Comment by cpanato** [comment]
retest this please on azure

**Comment by cpanato** [comment]
retest this please

File: tests/rspec/lib/cluster.rb [hunk 1]
```diff
@@ -118,8 +118,8 @@ def api_ip_addresses
       .map { |pod| [pod[0], nodes[pod[1]]] }.to_h
   end
 
-  def forensic
-    save_kubernetes_events(@kubeconfig, @name)
+  def forensic(events = true)
```
Comment (enxebre) [inline tests/rspec/lib/cluster.rb]: could also make events default to false as I see two calls not using true here. Nevertheless seems a reasonable pattern to have all enable by default for now

File: tests/rspec/lib/cluster.rb [hunk 2]
```diff
@@ -256,7 +256,10 @@ def wait_for_service(service, ips)
       sleep 10
     end
 
-    raise "timeout waiting for #{service} service to bootstrap on any of: #{ips}"
+    puts 'Trying to collecting the logs...'
+    forensic(false) # Call forensic to collect logs when service timeout
+    msg = "timeout waiting for #{service} service to bootstrap on any of: #{ips}"
```
Comment (enxebre) [inline tests/rspec/lib/cluster.rb]: could just raise the string straightaway here

**Comment by enxebre** [comment]
lgtm

File: tests/rspec/lib/cluster.rb (hunk 2 above)
Comment (cpanato) [reply tests/rspec/lib/cluster.rb]: yes, i want to try this out, reverting

File: tests/rspec/lib/cluster.rb (hunk 1 above)
Comment (cpanato) [reply tests/rspec/lib/cluster.rb]: ok

**Comment by cpanato** [comment]
retest this please

**Comment by cpanato** [comment]
retest this please

**Pull request merged**